### DISCUSSION TITLE
Unification of controls naming e.g. Picker vs ComboBox

### DISCUSSION THREAD

**Discussion by McoreD** [dotnet/maui#4302] [Ideas]
Hi all,

How’s the discussion on unification of controls naming going? In Xamarin we have **Picker**, but in WPF we call it **ComboBox**. Would a common name be chosen to handle both frameworks in MAUI?

It would make sense to unify not just control names, but also methods. 

Thanks,
Michael

**Comment by hartez** [comment]
> In Xamarin we have Picker, but in WPF we call it ComboBox. 

A Picker and a ComboBox aren't really the same thing. Pickers are limited to a pre-set list of items to choose from; the analog from WPF would be a ListBox. ComboBox has other features (e.g. text entry) that aren't available in Picker.

And it's unlikely we would change the name in MAUI from Picker to ListBox, because that name doesn't accurately describe the feature as it manifests on every platform. Depending on the rendering platform, a Picker might manifest as a dropdown list, a pop-up list, a spinner, or something else. So we want the name to reflect what it's used for, rather than how it manifests. (We don't always succeed in that, but it's the goal.)

Also, changing the names is a breaking change for Forms users who are migrating to MAUI, so we have limited the number of control name changes.
